apigw-rest-api-dynamodb-lambda-cdk/README.md

```diff
@@ -0,0 +1,144 @@
+
+## Requirements
+
+* [Create an AWS account](https://portal.aws.amazon.com/gp/aws/developer/registration/index.html) if you do not already have one and log in. The IAM user that you use must have sufficient permissions to make necessary AWS service calls and manage AWS resources.
+* [AWS CLI](https://docs.aws.amazon.com/cli/latest/userguide/install-cliv2.html) installed and configured
+* [Git Installed](https://git-scm.com/book/en/v2/Getting-Started-Installing-Git)
+* [AWS Serverless Application Model](https://docs.aws.amazon.com/serverless-application-model/latest/developerguide/serverless-sam-cli-install.html) (AWS SAM) installed
+
+## Architecture
+The following diagram illustrates the solutions architect
+![Architecture Diagram](img/concept.png)
+## Deployment Instructions
+
+1. Create a new directory, navigate to that directory in a terminal and clone the GitHub repository:
+    ``` 
+    git clone https://github.com/aws-samples/serverless-patterns
+    ```
+1. Change directory to the pattern directory:
+    ```
+    cd apigw-rest-api-dynamodb-lambda-cdk/cdk
+    ```
+
+1. Install dependencies
+    ```
+    npm install
+	cd src
+	pip install aws-xray-sdk --target .
+	pip install aws_lambda_powertools --target .
+    ```
+
+1. Deploy the stack to your default AWS account and region. The output of this command should give you the HTTP API URL.
+    ```
+    cdk deploy
+    ```
+
+## How it works
+
+This pattern creates an Amazon API Gateway REST API that integrates with an Amazon DynamoDB table. The API integrates directly with the DynamoDB API and supports [PutItem](https://docs.aws.amazon.com/amazondynamodb/latest/APIReference/API_PutItem.html) and [Query](https://docs.aws.amazon.com/amazondynamodb/latest/APIReference/API_Query.html) actions. The API Integration has read and write access to the DynamoDB. There is no authentication on the API endpoint
+
+## Testing
+
+Upon deployment, you will see the API endpoint URL in the output. It will take the format:
+
+`https://${API_ID}.execute-api.${REGION_NAME}.amazonaws.com`
+
+Once the application is deployed, use [Postman](https://www.postman.com/) to test the API using the following instructions.
+
+1. Launch Postman
+
+1. Invoke the DynamoDB **PutItem** action to add a new item to the DynamoDB table:
+	* Enter the API URL with the **prod** stage as the path:.
+	```
+	https://${API_ID}.execute-api.${REGION_NAME}.amazonaws.com/prod/awsomedynamodb
+	```
+	* Select **POST** as the HTTP method from the drop-down list to the left of the address bar.
+	* Choose the **Body** tab. Choose **raw** and select **JSON** from the drop-down list. Enter the following into the text box: 
+	```
+	{
+		"id":"14",
+    	"name":"Do some thing",
+    	"description":"Do some thing",
+    	"customer":"John"
+	}
```

Comment:
Th items you are adding don't seem relevant to what you query in the testing

Reply:
I am not sure about this, you can put anything for testing, right ?

Reply:
Yes, you can put anything. But although DynamoDB support schemeless, we actually should control scheme in real product